java/src/main/java/com/sinch/rtc/examples/hms/HmsOAuthFlowTokenValidator.java

```diff
@@ -79,7 +79,23 @@ public boolean IsTokenValid(String clientAssertionJwt, OffsetDateTime now) {
 
     Claims claims = jwt.getBody();
 
-    return "https://push-api.cloud.huawei.com".equals(claims.get("scope"));
+    if (!"https://push-api.cloud.huawei.com".equals(claims.get("scope"))) {
+      return TokenValidationResult.Invalid;
+    }
+
+    // Extract Sinch Application Key from JWT header parameter, and HMS Application ID from JWT
+    // claims.
+    return TokenValidationResult.Valid(
+        getSinchApplicationKey(jwt.getHeader()), getHmsApplicationId(claims));
```

Comment:
If I get it right, "sinch:rtc:application_key" being added to the header is never validated, which might result in it being `null` or corrupt. Consider adding additional check vs (at this point) already validated `iss` field from the payload.

Reply:
At this point, the signature of the JWT has already been validated, by means of using the _Application Key_ in the header to lookup corresponding _Application Secret_ and then deriving the signing key, so I'd say at this point this is safe to access the header.

Re `iss`, we actually don't use it to verify the signature. One reason is the `iss` has a more complex format than the `sinch:rtc:application_key` and extracting the _Application Key_ from `iss` would require parsing the still unvalidated JWT payload, which is not something we should do or encourage. Hence the _Application Key_ in the JWT header.

Reply:
Though this raises whether we should add back the `sinch:rtc:application_key` as a claim (as in the original proposal). So despite it may look "confusing" that the app key is present in the header and the payload, it is safer in a sense that once the signature has been verified, the _Sinch Application Key_ should be extracted from the (signed) _claim_, and not go back to the JWT header.